lib/api/root.rb

```diff
@@ -59,19 +63,32 @@ def call(object, _env)
 
     parser :json, Parser.new
 
+    use OpenProject::Authentication::Manager
+
     helpers do
       def current_user
-        return User.current if running_in_test_env?
-        user_id = env['rack.session']['user_id']
-        User.current = user_id ? User.find(user_id) : User.anonymous
+        User.current
+      end
+
+      def warden
+        env['warden']
       end
 
       def authenticate
-        if Setting.login_required? && (current_user.nil? || current_user.anonymous?)
+        warden.authenticate! scope: API_V3
+
+        User.current = warden.user scope: API_V3
+
+        if Setting.login_required? && not_logged_in?
           raise API::Errors::Unauthenticated
         end
       end
 
+      def not_logged_in?
```

Comment:
Can we please avoid using negated bool-methods? (when there is no existing positive method)

Soon someone will write `!not_logged_in?`, because the `not_logged_in?` method was already available...